### ISSUE TITLE
Remove InfixDirection from Operation

### ISSUE THREAD

**Issue by jfmengels** [stil4m/elm-syntax#195]
When we pattern match on Operator (`Operator "+" infixDirection left right ->`) the `InfixDirection` type is stored to describe something that is not present in the AST. I suggest we remove it.

In practice, I don't think this is useful at all. If we do need to know the infix direction (to rebalance an AST for instance), we can look it up, as these values are known based on the operator.

**Comment by MartinSStewart** [comment]
My understanding is that infix direction is only there because it's used in some intermediate parsing step. We should definitely remove it.

**Comment by jfmengels** [comment]
Yeah, I think it's only used in post-processing. We could use a lookup table instead, though ideally we remove the post-processing entirely and parse these expressions in the correct way the first time (using https://package.elm-lang.org/packages/dmy/elm-pratt-parser/latest/ for instance).

(I actually have started a branch for this, I should create an issue and a WIP draft.)

**Comment by jfmengels** [comment]
One thing that we could also do is move the arguments:
```elm
Operator "+" left right ->
-->
Operator left "+" right ->
```

I think this would be a bit more obvious, no?
